### DISCUSSION TITLE
The workbench failed to connect to the server (Error: WebSocket close with status code 1006)

### DISCUSSION THREAD

**Discussion by CoBool** [coder/code-server#4886] [Q&A] [answered]
system

Ubuntu server 20.04 LTS
Apache2

![image](https://user-images.githubusercontent.com/96224373/154798513-128694bc-38c4-4ffe-8363-5fa8daaee503.png)

config/code-server/config.yaml

bind-addr: IP:Port
auth: password
password: pass
cert: false

/etc/apache2/sites-available/code-server

<VirtualHost *:80>
    ServerName URL

    <IfModule mod_rewrite.c>
    RewriteEngine on

    RewriteRule ^ - [E=protossl]
    RewriteCond %{HTTPS} on
    RewriteRule ^ - [E=protossl:s]

    RewriteCond %{HTTPS} !=on
    RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

    </IfModule>

</VirtualHost>

# Specify the SSL cache directory. If possible, use shmcb, otherwise use the provided path.
SSLStaplingCache shmcb:/var/run/ocsp(128000)

<VirtualHost *:443>
    ServerName code.URL

    ErrorLog ${APACHE_LOG_DIR}/URL-error.log
    CustomLog ${APACHE_LOG_DIR}/URL-access.log combined

    Header always set Strict-Transport-Security "max-age=31536000"

    SSLEngine on

    SSLProtocol all -SSLv2 -SSLv3

    SSLCipherSuite ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA

    SSLHonorCipherOrder on
    SSLCertificateFile "/etc/letsencrypt/live/URL/cert.pem"
    SSLCertificateKeyFile "/etc/letsencrypt/live/URL/privkey.pem"
    SSLCertificateChainFile "/etc/letsencrypt/live/URL/chain.pem"

    # Turn on OCSP stapling
    SSLUseStapling on
    SSLStaplingResponderTimeout 5
    SSLStaplingReturnResponderErrors off

    RewriteEngine On
    RewriteCond %{HTTP:Upgrade} =websocket [NC]
    RewriteRule /(.*)           wss://ip:port/$1 [P,L]
    RewriteCond %{HTTP:Upgrade} !=websocket [NC]
    RewriteRule /(.*)           http://ip:port/$1 [P,L]
</VirtualHost>

I set it like this, but I keep getting an error 1006.

sorry for my english

**Comment by cxw620** [comment, marked as answer]
Using CDN? Some CDN don't support websocket and I met with same problem, for I used qiniu dynamic CDN at first and it didn't support websocket. Later I make use of ECDN of Tencent CDN and it works. You can try to connect to code-server with ip:port directly, if problem still exists, code-server itself may be to blame and you can open an issue. If not, check your apache config refering to official docs.
I notice this `SSLProtocol all -SSLv2 -SSLv3`.  Try to change it to `SSLProtocol +TLSv1.2`?

**Reply by CoBool** [reply]
I solved it by reinstalling version 3.14

I'll change it the way I told you next time!

thanks
